File: scripts/extract_pgi_wines.mjs
import { readFileSync, writeFileSync } from 'fs';

const raw = readFileSync('C:/Users/neilw/.claude/projects/C--Users-neilw-Documents-GitHub-loamv2/fe51cf62-7e9b-4cc6-b236-7f3c2be9d97e/tool-results/bh721bw84.txt', 'utf8');
const data = JSON.parse(raw);

const pgiWines = data.filter(d => d.giType === 'PGI' && d.productType === 'WINE' && d.status === 'registered');

const extracted = pgiWines.map(d => ({
  name: d.protectedNames[0],
  file_number: d.fileNumber,
  country: d.countries[0],
  eu_protection_date: d.euProtectionDate,
  gi_identifier: d.giIdentifier,
  transcriptions: d.transcriptions || null,
})).sort((a, b) => {
  if (a.country !== b.country) return a.country.localeCompare(b.country);
  return a.name.localeCompare(b.name);
});

writeFileSync('data/eambrosia_pgi_wines.json', JSON.stringify(extracted, null, 2));
console.log('Saved', extracted.length, 'PGI wines to data/eambrosia_pgi_wines.json');

// Show by country
for (const country of ['IT', 'FR', 'ES', 'PT', 'DE', 'GR', 'AT', 'RO', 'HU', 'SI', 'BG', 'CZ', 'CY', 'GB', 'NL', 'DK', 'BE', 'MT', 'SK', 'CN', 'US']) {
  const entries = extracted.filter(e => e.country === country);
  if (entries.length === 0) continue;
  console.log(`\n--- ${country} (${entries.length}) ---`);
  entries.forEach(e => console.log(`  ${e.name}`));
}

File: pipeline/reference/extract_pgi_wines.py
"""
Filter PGI wine entries from eAmbrosia JSON data.

Reads the full eAmbrosia GI dataset, filters to PGI + WINE + registered,
extracts key fields, and writes to data/eambrosia_pgi_wines.json.

Usage:
    python -m pipeline.reference.extract_pgi_wines <input.json>
"""

import argparse
import json
import sys
from pathlib import Path

# Allow running as module from repo root
sys.path.insert(0, str(Path(__file__).resolve().parents[2]))

PROJECT_ROOT = Path(__file__).resolve().parents[2]


def main():
    parser = argparse.ArgumentParser(description="Extract PGI wines from eAmbrosia JSON")
    parser.add_argument("input_file", help="Input JSON file (full eAmbrosia dataset)")
    parser.add_argument("--output", default=str(PROJECT_ROOT / "data" / "eambrosia_pgi_wines.json"),
                        help="Output JSON file")
    args = parser.parse_args()

    raw = Path(args.input_file).read_text(encoding="utf-8")
    data = json.loads(raw)

    pgi_wines = [
        d for d in data
        if d.get("giType") == "PGI"
        and d.get("productType") == "WINE"
        and d.get("status") == "registered"
    ]

    extracted = sorted(
        [
            {
                "name": d["protectedNames"][0],
                "file_number": d.get("fileNumber"),
                "country": d["countries"][0],
                "eu_protection_date": d.get("euProtectionDate"),
                "gi_identifier": d.get("giIdentifier"),
                "transcriptions": d.get("transcriptions") or None,
            }
            for d in pgi_wines
        ],
        key=lambda x: (x["country"], x["name"]),
    )

    out_path = Path(args.output)
    out_path.write_text(json.dumps(extracted, indent=2, ensure_ascii=False), encoding="utf-8")
    print(f"Saved {len(extracted)} PGI wines to {out_path}")

    # Show by country
    show_countries = [
        "IT", "FR", "ES", "PT", "DE", "GR", "AT", "RO", "HU", "SI",
        "BG", "CZ", "CY", "GB", "NL", "DK", "BE", "MT", "SK", "CN", "US",
    ]
    for country in show_countries:
        entries = [e for e in extracted if e["country"] == country]
        if not entries:
            continue
        print(f"\n--- {country} ({len(entries)}) ---")
        for e in entries:
            print(f"  {e['name']}")


if __name__ == "__main__":
    main()
